### PR TITLE
5.x.x/requests

### PR DESCRIPTION
**Pull request by Pjiesco**
Added Requests:
- GetTransitionSettings
- SetCurrentTransitionDuration
- SetCurrentTransition
- SetTransitionSettings
- GetTransitionSettings 
- ReleaseTbar
- SetTbarPosition
- TriggerStudioModeTransition
- GetSourceFilter
- SetSourceFilterIndex
- GetSourceFilterList
- CreateSourceFilter
- RemoveSourceFilter
- SetSourceFilterEnabled
- SetSourceFilterSettings
- GetSceneItemList
- GetSceneItemEnabled
- SetSceneItemEnabled
- GetSceneItemLocked
- CreateSceneItem
- DuplicateSceneItem
- GetSceneItemColor
- RemoveSceneItem
- SetSceneItemIndex
- SetSceneItemLocked

### CODE REVIEW

File: src/main/java/net/twasi/obsremotejava/message/response/sceneItems/GetSceneItemColorResponse.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import net.twasi.obsremotejava.message.request.Request;
+import net.twasi.obsremotejava.message.response.RequestResponse;
+
+@Getter
+@ToString(callSuper = true)
+public class GetSceneItemColorResponse extends RequestResponse {
+    private Data responseData;
+
+    public GetSceneItemColorResponse() {
+        super(Request.Type.SetSceneItemLocked);
```
Comment (ChristopheCVB) [inline src/main/java/net/twasi/obsremotejava/message/response/sceneItems/GetSceneItemColorResponse.java:14]: Wrong type

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]
Amazing work here! A lot has been done, maybe too much for 1 PR though ;)

File: src/main/java/net/twasi/obsremotejava/message/event/sceneitems/SceneItemSourceEvent.java [hunk 2]
```diff
@@ -14,6 +14,6 @@ protected SceneItemSourceEvent(Type eventType, Category category) {
     @ToString(callSuper = true)
     protected static class Data extends SceneItemEvent.Data {
         private String sourceName;
-        private String sourceItemId;
+        private String sourceItemId; // String or Integer?
```
[the remark this answers is not in the document]
Comment (Pjiesco) [reply src/main/java/net/twasi/obsremotejava/message/event/sceneitems/SceneItemSourceEvent.java:17]: We actually know that this is a Integer just looks like i missed this one while setting them all to a Integer

**Comment by Pjiesco** [comment]
@ChristopheCVB Yeah, i also found it a bit to much. I'm planning on working per category now

**Review by ChristopheCVB — APPROVED** [review]

**Pull request merged**